pkgs/build-support/rust/import-cargo-lock.nix

```diff
@@ -0,0 +1,167 @@
+{ fetchgit, fetchurl, lib, runCommand, cargo, jq }:
+
+{
+  # Cargo lock file
+  lockFile
+
+  # Hashes for git dependencies.
+, outputHash ? {}
+}:
+
+let
+  # Parse a git source into different components.
+  parseGit = src:
+    let
+      parts = builtins.match ''git\+([^?]+)(\?rev=(.*))?#(.*)?'' src;
+      rev = builtins.elemAt parts 2;
+    in
+      if parts == null then null
```

Comment:
IIUC, this could mean `parseGit foo` evaluates to `null`, right? We use it below as `gitParts.sha` so wouldn't this crash at runtime?